src/graph.h

```diff
@@ -250,6 +250,10 @@ struct Edge {
   bool is_phony() const;
   bool use_console() const;
   bool maybe_phonycycle_diagnostic() const;
+
+  // Historical info: how long did this edge take last time,
+  // as per .ninja_log, if known? Defaults to -1 if unknown.
+  int64_t prev_elapsed_time;
```

Comment:
Consider instead using std::chrono so that you can represent actual time values, and not just only, well, whatever unit this is in. Seconds?

Reply:
Personally I'd *really* like to see std::chrono used, but it's not my call for ninja.

If this PR gets merged, i'll replace your time keeping type with std::chrono in my fork/patch series though (splinter-build, on github), because I'd like to have this change, but would rather have std::chrono :-)

Reply:
Yep, naked unitless data is a problem, i agree. I would think it would make sense to change
all relevant occurrences at once, so i'm not sure an one-off change is a win?

Reply:
Well this is a new feature, and i don't see any documentation in the current code for what `prev_elapsed_time` represents in terms of units.

So for this new code, it should probably be std::chrono, and if you need to adapt from something that's already present, just adapt in the few places where that needs to happen.

Other raw-integer time variables can be adjusted later by someone else.

Reply:
My main problem with that is that it (and some other changes) goes directly against the current code style,
and having uniform code is generally better, and can be uplifted all at once,
as opposed to having a weird mismatch of different coding styles.

But if the maintainers (@jhasse ? @nico ?) can chime in and review the change as a whole,
i can sure make those changes if they are deemed better by maintainers.

Reply:
There's precedent that "update a specific thing for the whole codebase" will be rejected, and only "while we're already modifying / adding, do it 'correctly'" will be accepted.

But whatever you want to do. Not my call.